enhancements/network/infiniband.md

```diff
@@ -0,0 +1,123 @@
+---
+title: infiniband
+authors:
+  - "@zshi"
+reviewers:
+  - "@pliurh"
+  - "@dougbtv"
+  - "@s1061123"
+  - "@danwinship"
+  - "@squeed"
+  - "@dcbw"
+  - "@yrobla"
+approvers:
+  - "@knobunc"
+  - "@fepan"
+creation-date: 2020-05-06
+last-updated: 2020-05-06
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Infiniband
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RDMA (Remote Direct Memory Access) fabrics such as Infiniband and RDMA over
+Converged Ethernet (RoCE) are key to unlocking scalable performance for
+demanding Artificial Intelligence (AI) and Machine Learning (ML) applications.
+RoCE v2 is supported in SR-IOV Operator as Tech Preview since 4.3, This
+enhancement is to introduce native Infiniband support.
+
+## Motivation
+
+Enable high-performance message transmission via RDMA for distributed workloads
+such as AI/ML.
+
+### Goals
+
+- Provide native Infiniband support on pod additional network interfaces
+
+### Non-Goals
+
+- Support RoCE v2
+- Support GPUDirect RDMA
+- Support Ethernet over Infiniband such as IPoIB (IP over Infiniband)
```

Comment:
by using SR-IOV the VF will use the ipoib netdevice.

Reply:
@moshe010  Could you elabrate it? do you mean current `ib-sriov-cni` only creates ipoib netdevice?

Reply:
No, for the feature to work ib_ipoib need to be loaded to expose the VF netdevice.  we don't create ipoib netdevice.

Reply:
ok, from the rdma-core package, it seems indicating that `ib_ipoib` module is loaded by the system if any InfiniBand device is installed.

I didn't quite understand why ipoib driver is needed for ib only device. maybe the ipoib driver contains functions that also apply to ib device?

Reply:
I dont think there is a standard way to configure Port GUID of a VF except via ipoib interface of the PF.

@moshe010 do you know ?